docs/INTEGRATION_TESTS.md

```diff
@@ -0,0 +1,96 @@
+            // URL to fetch a timeseries from egress based on station_id and param_id, with our
+            // station and param templated in
+            let url = format!(
+                "http://localhost:3000/stations/{}/params/{}",
+                ts.station_id, param.id
+            );
+            // GET that URL
+            let resp = reqwest::get(url).await.unwrap();
+            // Assert sucess response from egress 
+            assert!(resp.status().is_success());
+        }
+    })
+    .await
+}
+```
+
+## Understanding e2e_test_wrapper
+
+`e2e_test_wrapper` makes it easier to write integration tests by containing the boilerplate of setting up the services that comprise lard, and making them available to a closure, then cleaning up the database afterward. It does this setup by starting the HTTP servers as tokio tasks instead of independent processes:
+```rust
+let egress_server = tokio::spawn(async move {
+    tokio::select! {
+        output = lard_egress::run(api_pool, cancel_token2) => output,
+        _ = init_shutdown_rx1.recv() => {
+            api_shutdown_tx.send(()).unwrap();
+        },
+    }
+});
+```
+
+It then selects across the join handles of the services, and the future of closure passed in:
+```rust
+tokio::select! {
+    _ = egress_server => panic!("API server task terminated first"),
+    _ = ingestor => panic!("Ingestor server task terminated first"),
+    _ = sig_catcher => panic!("Signal catcher caught a shutdown signal"),
+    // Clean up database even if test panics, to avoid test poisoning
+    test_result = AssertUnwindSafe(test).catch_unwind() => {
+        // For debugging a specific test, it might be useful to skip the cleanup process
+        #[cfg(not(feature = "debug"))]
+        for db_pool in [open_db_pool, restricted_db_pool] {
+            let client = db_pool.get().await.unwrap();
+            client
+                .batch_execute(
+                    // TODO: should clean public.timeseries_id_seq too? RESTART IDENTITY CASCADE?
+                    "TRUNCATE public.timeseries, labels.met, labels.obsinn CASCADE",
+                )
+                .await
+                .unwrap();
+        }
+        assert!(test_result.is_ok())
+    }
+}
+```
+Here we expect the test_result to complete first, as the server tasks should run indefinitely unless they crash or until we tell them to shut down, so we panic and fail the test if this does not happen.
+
+In the handler for the test closure, we make sure to clean up the database before moving on, so the next test can start on a blank slate. We make sure to catch any panics so we will perform this cleanup even if the closure fails.
+
+It's worth noting that e2e_test_wrapper does not set up a postgres instance, just connects to it, so a postgres instance must be available to run these tests. To help with this, we've set up a justfile to use instead of running `cargo test` directly, which sets up a postgres instance in a docker container.
```

Comment:
```suggestion
It's worth noting that `e2e_test_wrapper` does not set up a postgres instance, just connects to it, so a postgres instance must be available to run these tests. To help with this, we've set up a justfile to use instead of running `cargo test` directly, which sets up a postgres instance in a docker container.
```

Reply:
Fixed